src/traces/waterfall/attributes.js

```diff
@@ -78,9 +80,9 @@ module.exports = {
 
     orientation: barAttrs.orientation,
 
-    offset: barAttrs.offset,
+    offset: extendFlat({}, barAttrs.offset, { arrayOk: false }),
```

Comment:
They used to work on separated bars. But then I noticed it would become difficult to read the `waterfall` graph and more complicated to compute the connector endpoints.

Reply:
So, they _still_ work and you made them `arrayOk:false` to avoid confusion?

Reply:
I think that arrayOk `offset` and `width` are still valuable, especially if users want something more general than the set [waterfallmode, waterfallgap and waterfallgroupgap].

If things get too messy bringing back arrayOk `offset` and `width`, I would prefer :hocho: `offset` and `width` entirely from this first iteration than having them in but as `arrayOk:false`.

Reply:
Thanks for the comments @etpinard. OK I would try to keep them there.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3531/commits/a05352fa492c89dae5cde3b2afbae445811750fe.

Reply:
Looking good. Thanks!